xc7/primitives/slicem/Ndram/ntemplate.N_dram.pb_type.xml

```diff
@@ -14,50 +14,20 @@
 
   <!-- TODO: Missing modes: SRL -->
   <mode name="LUT">
-    <pb_type name="{N}5LUT" num_pb="2" class="lut" blif_model=".names">
-      <input  name="in"  num_pins="5" port_class="lut_in" />
-      <output name="out" num_pins="1" port_class="lut_out" />
-      <delay_matrix type="max" in_port="in" out_port="out">
-        {{iopath_A1_O5}}
-        {{iopath_A2_O5}}
-        {{iopath_A3_O5}}
-        {{iopath_A4_O5}}
-        {{iopath_A5_O5}}
-      </delay_matrix>
-      <metadata>
-        <meta name="type">bel</meta>
-        <meta name="subtype">lut</meta>
-      </metadata>
-    </pb_type>
-    <xi:include href="../../common_slice/muxes/f6mux/f6mux.pb_type.xml" />
-
+    <xi:include href="../../common_slice/Nlut/ntemplate.Nlut.pb_type.xml" />
     <interconnect>
-      <!-- LUT5 (upper) -> O6 -->
-      <direct name="{N}LUT_A5_0" input="{N}_DRAM.A[4:0]" output="{N}5LUT[0].in[4:0]"/>
-
-      <!-- LUT5 (lower) -> O5 -->
-      <direct name="{N}LUT_A5_1" input="{N}_DRAM.A[4:0]" output="{N}5LUT[1].in[4:0]"/>
+      <direct name="A1" input="{N}_DRAM.A[0]" output="{N}LUT.A1"/>
+      <direct name="A2" input="{N}_DRAM.A[1]" output="{N}LUT.A2"/>
+      <direct name="A3" input="{N}_DRAM.A[2]" output="{N}LUT.A3"/>
+      <direct name="A4" input="{N}_DRAM.A[3]" output="{N}LUT.A4"/>
+      <direct name="A5" input="{N}_DRAM.A[4]" output="{N}LUT.A5"/>
+      <direct name="A6" input="{N}_DRAM.A[5]" output="{N}LUT.A6"/>
 
-      <!-- MUX used for LUT6 -->
-      <direct name="F6MUX_I0" input="{N}5LUT[0].out" output="F6MUX.I0"><pack_pattern name="LUT5toLUT6"/></direct>
-      <direct name="F6MUX_I1" input="{N}5LUT[1].out" output="F6MUX.I1"><pack_pattern name="LUT5toLUT6"/></direct>
-      <direct name="F6MUX_S"  input="{N}_DRAM.A[5]"  output="F6MUX.S"></direct>
-
-      <!-- LUT outputs -->
-      <direct name="O5" input="{N}5LUT[0].out" output="{N}_DRAM.O5">
-        <pack_pattern name="LUT5x2"/>
-      </direct>
-      <mux    name="O6" input="{N}5LUT[1].out F6MUX.O" output="{N}_DRAM.O6">
-        <pack_pattern in_port="F6MUX.O"        name="LUT5toLUT6" out_port="{N}_DRAM.O6"/>
-        <pack_pattern in_port="{N}5LUT[1].out" name="LUT5x2"     out_port="{N}_DRAM.O6"/>
-      </mux>
+      <direct name="O5" input="{N}LUT.O5" output="{N}_DRAM.O5"/>
+      <direct name="O6" input="{N}LUT.O6" output="{N}_DRAM.O6"/>
     </interconnect>
     <metadata>
-      <meta name="fasm_type">SPLIT_LUT</meta>
-      <meta name="fasm_lut">
-        INIT[31:0] = {N}5LUT[0]
-        INIT[63:32] = {N}5LUT[1]
-      </meta>
+      <meta name="fasm_prefix">d{N}LUT</meta>
```

Comment:
@mithro is the `d` previous to {N}LUT necessary? Because it creates wrong `.fasm` output. I have run some `dram` tests after removing `d, b and c` from the `fasm_prefixes` and the issue seems to be solved. I am also running the `all_xc7` tests to see if the issue is present somewhere else, and the `dram_1_32m` correctly completes.

Reply:
@acomodi - The issue I'm seeing is that we are getting **both** the `dCLUT` and `bCLUT` output for the **same** BEL -- this shouldn't be possible as the `dCLUT` and the `bCLUT` tags are on the different modes of the same block.
```
Segment DB CLBLM_L, key CLBLM_L.SLICEM_X0.dCLUT.RAM not found from line 'CLBLM_L_X20Y110.SLICEM_X0.dCLUT.RAM'                                        
Segment DB CLBLM_L, key CLBLM_L.SLICEM_X0.dCLUT.SMALL not found from line 'CLBLM_L_X20Y110.SLICEM_X0.dCLUT.SMALL'                                                                            
Segment DB CLBLM_L, key CLBLM_L.SLICEM_X0.bCLUT.RAM not found from line 'CLBLM_L_X20Y110.SLICEM_X0.bCLUT.RAM'                                                                                
Segment DB CLBLM_L, key CLBLM_L.SLICEM_X0.bCLUT.SMALL not found from line 'CLBLM_L_X20Y110.SLICEM_X0.bCLUT.SMALL' 
```

Reply:
The actual FASM tags which are output is not important, the important fact is that it seems to be producing output for an impossible case.

Reply:
Ok, so the addition of `d, c, b` was just for debugging purposes. And the issue here is that a same bel is expressed in two different modes, which, as you say, is an impossible case. I will double-check whether the `.net` file produced after packing is correct, to be sure that the issue is indeed in `fasm.cpp`.

Reply:
@acomodi Correct!